gwcli/ceph.py

```diff
@@ -285,14 +287,20 @@ class RadosPool(UINode):
     display_attributes = ["name", "commit", "overcommit_PCT", "max_bytes",
                           "used_bytes", "type", "desc"]
 
-    def __init__(self, parent, pool_name, pool_md):
+    def __init__(self, parent, pool_name, pool_md, ecprofiles):
         UINode.__init__(self, pool_name, parent)
         self.pool_md = pool_md
+
+        if self.pool_md['type'] == 3:
+            k = ecprofiles[self.pool_md['erasure_code_profile']]['k']
+            m = ecprofiles[self.pool_md['erasure_code_profile']]['m']
+        else:
+            k = 'N'
+            m = 'N'
+
         pool_type = {1: ("x{}".format(self.pool_md['size']),
                          'replicated'),
-                     3: ("{}+{}".format(self.pool_md['min_size'],
-                                        self.pool_md['size'] - self.pool_md['min_size']),
-                         "erasure")}
+                     3: ("{}+{}".format(k, m), "erasure")}
```

Comment:
This `pool_type` dict and initialization of `k` and `m` to `N` seem redundant to me.  How about something like:
```
type_id = pool_md['type']
if type_id == 1:
    self.desc = "x{}".format(pool_md['size'])
    self.type = "replicated"
elif type_id == 3:
    ec_profile = ec_profiles[pool_md['erasure_code_profile']]
    self.desc = "{}+{}".format(ec_profile['k'], ec_profile['m'])
    self.type = "erasure"
```
and dropping `pool_type` and `self.pool_md`?

Reply:
Yeah, this looks better, will fix it.